nomenclature/iamc/datapoint.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+from pydantic import BaseModel, field_validator
+
+
+from nomenclature.definition import DataStructureDefinition
+
+IAMC_DIMENSIONS = ["model", "scenario", "region", "variable", "unit"]
+
+
+class IamcDataFilter(BaseModel):
+    model: List[str] | None = None
+    scenario: List[str] | None = None
+    region: List[str] | None = None
+    variable: List[str] | None = None
+    unit: List[str] | None = None
+    year: List[int] | None = None
+
+    @field_validator(*IAMC_DIMENSIONS, mode="before")
+    @classmethod
+    def single_input_to_list(cls, v):
+        return v if isinstance(v, list) else [v]
+
+    def validate(self, dsd: DataStructureDefinition) -> None:
```

Comment:
The `validate` is already defined from `BaseModel` and overwriting it would break the pydantic validation. You'll have to chose a different name. On my local machine mypy warns about this.

Reply:
Rename to `validate_with_definition()` in parallel to the `RequiredData` class.